### PR TITLE
Java: Added Hget, Hset and Hdel to BaseClient and BaseTransaction. (Hash Commands)

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

File: java/client/src/test/java/glide/api/RedisClientTest.java [hunk 1]
```diff
@@ -201,4 +207,74 @@ public void set_with_SetOptions_OnlyIfDoesNotExist_returns_success() {
...
+        // setup
+        String key = "testKey";
+        Map<String, String> fieldValueMap = Map.of("field1", "value1", "field2", "value2");
+        String[] args = new String[] {key, "field1", "value1", "field2", "value2"};
+        Long value = 2L;
+
+        CompletableFuture testResponse = mock(CompletableFuture.class);
+        when(testResponse.get()).thenReturn(value);
+        when(commandManager.<Long>submitNewCommand(
+                        eq(HashSet), MockitoHamcrest.argThat(arrayContainingInAnyOrder(args)), any()))
```
Comment (Yury-Fridlyand) [inline java/client/src/test/java/glide/api/RedisClientTest.java:246]: You can use `LinkedHashMap` for fixed order

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/HashCommands.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     * @see <a href="https://redis.io/commands/hget/">redis.io</a> for details.
+     * @param key The key of the hash.
+     * @param field The field in the hash stored at <code>key</code> to retrieve from the database.
+     * @return The value associated with <code>field</code>, or <code>null</code> when <code>field
+     *     </code> is not present in the hash or <code>key</code> does not exist.
+     */
+    CompletableFuture<String> hget(String key, String field);
+
+    /**
+     * Sets the specified fields to their respective values in the hash stored at <code>key</code>.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/HashCommands.java:26]: We need to keep tense consistent (even if other clients are not)
```suggestion
     * Set the specified fields to their respective values in the hash stored at <code>key</code>.
```

File: java/client/src/main/java/glide/api/commands/HashCommands.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+     * @see <a href="https://redis.io/commands/hset/">redis.io</a> for details.
+     * @param key The key of the hash.
+     * @param fieldValueMap A field-value map consisting of fields and their corresponding values to
+     *     be set in the hash stored at the specified key.
+     * @return The number of fields that were added.
+     */
+    CompletableFuture<Long> hset(String key, Map<String, String> fieldValueMap);
+
+    /**
+     * Removes the specified fields from the hash stored at <code>key</code>. Specified fields that do
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/HashCommands.java:37]:
```suggestion
     * Remove the specified fields from the hash stored at <code>key</code>. Specified fields that do
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 4]
```diff
@@ -144,6 +150,63 @@ public T set(String key, String value) {
...
+     */
+    public T hget(@NonNull String key, @NonNull String field) {
+        ArgsArray commandArgs = buildArgs(key, field);
+
+        protobufTransaction.addCommands(buildCommand(HashGet, commandArgs));
+        return getThis();
+    }
+
+    /**
+     * Sets the specified fields to their respective values in the hash stored at <code>key</code>.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:170]:
```suggestion
     * Set the specified fields to their respective values in the hash stored at <code>key</code>.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 5]
```diff
@@ -144,6 +150,63 @@ public T set(String key, String value) {
...
+                                        fieldValueMap.entrySet().stream()
+                                                .flatMap(entry -> Stream.of(entry.getKey(), entry.getValue())))
+                                .toArray(String[]::new));
+
+        protobufTransaction.addCommands(buildCommand(HashSet, commandArgs));
+        return getThis();
+    }
+
+    /**
+     * Removes the specified fields from the hash stored at <code>key</code>. Specified fields that do
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:192]:
```suggestion
     * Remove the specified fields from the hash stored at <code>key</code>. Specified fields that do
```

**Review by Yury-Fridlyand — COMMENTED** [review]
Please update transaction tests to include new commands

**Pull request merged**